first test for webWriter + .github changes + serverInfo struct w/ methods

gauron99 opened the pull request:
* created new tests for `webWriter()` 
* gh workflow running for basic tests 
@zroubalik

zroubalik commented on http-server/server/server.go:
```diff
@@ -11,17 +12,31 @@ import (
 var titles = []string{"the Mighty Traveler", "the Great Summoner", "the Conqueror of Titans", "the Destroyer of Worlds", "the King of Oceans", "the King of Underworld"}
 
+// GetParam returns pointer to param variable (for other packages)
+func GetParam() *string {
```
Would be better to return plain string, not pointer, see: https://dhdersch.github.io/golang/2016/01/23/golang-when-to-use-string-pointers.html

gauron99 replied:
rewrote using plain strings

zroubalik commented on http-server/server/server.go:
```diff
@@ -11,17 +12,31 @@ import (
+	if len(param) > 0 { //if a parameter is given, print it out everywhere where something is printed out
+
+		out := fmt.Sprintf("I have a parameter! Here: %s\n", param)
+		// outPrt := fmt.Sprintf("I have a parameter!! Here: %s\n", param)
+		// fmt.Fprintf(rw, outPrt)
```
is this needed?

gauron99 replied:
removed with latest commit

zroubalik commented on http-server/server/server.go:
```diff
@@ -11,17 +12,31 @@ import (
 )
 
 var pageAccessCount = 0
-var Param string
+var param string = ""
```
Would be better to convert this to a struct:
```go
type server struct (
  param string
  pageAccessCount int
}
```

And then create a methods on this struct.

zroubalik commented on http-server/server/server_test.go:
```diff
@@ -5,47 +5,86 @@ import (
+	//TODO test error case next
 }
+
+// TestApiHome test api call to home page "/"
+// It is expected that current time will be printed out
+// func TestApiHome(t *testing.T) {
+// 	req, err := http.NewRequest("GET", "/", nil)
+// 	if err != nil {
+// 		t.Fatal(err)
+// 	}
+// 	rr := httptest.NewRecorder()
+// 	hdlr := http.HandlerFunc(ApiHome)
+// 	hdlr.ServeHTTP(rr, req)
+
+// 	fmt.Println("status:", rr.Code)
+// 	if status := rr.Code; status != http.StatusOK {
+// 		t.Errorf("handler returned wrong status code: got %v want %v",
+// 			status, http.StatusOK)
+// 	}
+
+// 	expected := time.Now().Format("2006-01-02 15:04:05")
+// 	got := rr.Body.String()
+
+// 	fmt.Println("exp:", expected, "; got:", got)
+// 	if got != expected {
+
+// 		t.Errorf("Http body fail. Expected %v but got %v ", expected, got)
+// 	}
```
In general it is not a good practice to commit blocks of commented code.

gauron99 commented:
@zroubalik 
* rewrote server parameters into `struct`
* updated `GetParam` to be a method for this struct using plain string instead of pointer to string

gauron99 commented on http-server/server/server.go:
```diff
@@ -11,29 +11,34 @@ import (
+	param           string
+}
+
+var serv = serverInfo{}
```
`var Serv = serverInfo{}`
possibly have this public in order to use in `main` as `server.Serv`
then `SetParam()` can be a method as `func (si *serverInfo) SetParam(s string)`

gauron99 replied:
`var Serv = serverInfo{}`
possibly have this public in order to use in main as server.Serv
then `SetParam()` can be a method as `func (si *serverInfo) SetParam(s string)`

gauron99 commented on http-server/server/server.go:
```diff
@@ -11,29 +11,34 @@ import (
-func GetParam() *string {
-	return &param
+// SetParam takes string argument and sets it as parameter for the program
+func SetParam(s string) {
```
```
func (si *serverInfo) SetParam(s string) {
	si.param = s
}
```
reference the above comment

gauron99 commented on http-server/server/server.go:
```diff
@@ -10,18 +11,37 @@ import (
+	param           string
+}
+
+var serv = serverInfo{}
```
`var Serv = serverInfo{}`
possibly have this public in order to use in main as `server.Serv`
then `SetParam()` can be a method as `func (si *serverInfo) SetParam(s string)`

gauron99 commented on http-server/server/server.go:
```diff
@@ -10,18 +11,37 @@ import (
+// SetParam takes string argument and sets it as parameter for the program
+func SetParam(s string) {
+	serv.param = s
+}
```
```
func (si *serverInfo) SetParam(s string) {
	si.param = s
}
```
reference the comment above

gauron99 commented:
* made SetParam() as a method
* removed some useless comments

zroubalik commented on http-server/main.go:
```diff
@@ -27,13 +27,13 @@ func main() {
 	// handle possible cli arguments & set
 	flag.Parse()
-	server.Serv.SetParam(paramPtr)
+	serv := server.MakeServerInfo(0, "")
```
rename to `NewServerInfo` it's more idiomatic

zroubalik commented on http-server/main.go:
```diff
@@ -27,13 +27,13 @@ func main() {
 	flag.Parse()
-	server.Serv.SetParam(paramPtr)
+	serv := server.MakeServerInfo(0, "")
+	serv.Param = paramPtr
```
Why like this and not part of the function above?

zroubalik commented on http-server/server/server.go:
```diff
@@ -14,39 +14,32 @@ import (
 type serverInfo struct {
 	pageAccessCount int
-	param           string
+	Param           string
```
this don't have to be public, if you initiate it in contructor.

zroubalik reviewed (commented):
almost there :) No need to treat Param differently.

zroubalik commented on http-server/server/server.go:
```diff
@@ -10,25 +11,35 @@ import (
+}
+
+// MakeServerInfo creates & returns newly initialized serverInfo structure with given args
+func MakeServerInfo(pageAccessCnt int, parameter string) *serverInfo {
```
It kinda doesn't make a sense to set pageAccessCnt here via parameter.
Just set it init it in the constructor without param.

zroubalik reviewed (approved):
LGTM

zroubalik reviewed (approved):
LGTM :) ready to merge
